File: fishingclient/Assets/Scripts/Game/GameLogic/Player.cs
using UnityEngine;
using System.Collections;
using Fishing.Net;

namespace Fishing
{
    public class Player
    {
        private int _uID;
        private int _nGold;
        private int _nScore;
        private int _nLottery;
        private string _strNickName;
        private int _nPhotoId;
        private int _nLastDeltaLottery;
        public int UID
        {
            get { return _uID; }
            set { _uID = value; }
        }
        public int Gold
        {
            get { return _nGold; }
            set { _nGold = value; }
        }
        public int Score
        {
            get { return _nScore; }
            set { _nScore = value; }
        }
        public int Lottery
        {
            get { return _nLottery; }
            set { _nLottery = value; }
        }
        public string NickName
        {
            get { return _strNickName; }
            set { _strNickName = value; }
        }
        public int LastDeltaLottery
        {
            get { return _nLastDeltaLottery; }
            set { _nLastDeltaLottery = value; }
        }
        public int PhotoId
        {
            get { return _nPhotoId; }
            set { _nPhotoId = value; }
        }
        public Player()
        {

        }
        public Player(PlayerData playerData)
        {
            this._uID = playerData.uid;
            this._nGold = playerData.gold;
            this._nScore = playerData.score;
            this._nLottery = playerData.lottery;
            this._strNickName = playerData.nickName;
            this._nPhotoId = playerData.photoId;
        }
        public void Update(PlayerData playerData)
        {
            this._uID = playerData.uid;
            this._nGold = playerData.gold;
            this._nScore = playerData.score;
            this._nLottery = playerData.lottery;
            this._strNickName = playerData.nickName;
            this._nPhotoId = playerData.photoId;
        }
    }
}

File: fishing-server/game-server/app/domain/entity/player.js
/**
 * 
 * 玩家对象
 */
module.exports = function(opts) {
  return new player(opts);
};

var player = function(opts) {
  this.uid = opts.uid;                              //玩家id
  this.sid = opts.sid;                              //玩家连接的前端服务器id
  this.tableId = opts.tableId || 0;                 //玩家所在的频道(桌子)
  this.chairId = opts.chairId || 0;                 //玩家所在的椅子id
  this.gold = opts.gold || 0;                       //金币数量
  this.lottery = opts.lottery || 0;                 //彩票数量
  this.photoId = opts.photoId || 0;                 //头像
  this.level = opts.level || 0;                     //等级
  this.exp = opts.exp || 0;                         //经验
  this.nickName = opts.nickname;                    //昵称
  this.hook = opts.hook || 1;                       //钩子状态，默认为1
};